contracts/contracts/EventManager.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+contract EventManager {
+    struct Event {
+        uint256 eventId;
+        uint16 eventType;
+        bytes data;
+    }
+
+    mapping(uint => Event[]) private _eventsByType;
+    uint private _eventBufferSize;
+    address public writer;
+
+    constructor(uint bufferSize, address _writer) {
+        _eventBufferSize = bufferSize;
+        writer = _writer;
+    }
+
+    function insertEvent(uint256 eventId, uint16 eventType, bytes memory data) internal {
+        require(_eventsByType[eventType].length < _eventBufferSize, "Event buffer is full");
+        _eventsByType[eventType].push(Event(eventId, eventType, data));
+    }
+
+    function eraseEvents(uint256[] memory eventIds, uint16 eventType) external {
```

Comment:
how is `eraseEvents` used? i do not see any usages in `RandomnessGenerator.sol`

Reply:
also, do you think we need to accept a vector of IDs here? why a single ID insufficient?

Reply:
`eraseEvents` is for external usage :)
vector of IDS for batching

Reply:
why you can't use it inside `postRandomness`? i thought `postRandomness` + `eraseEvents` should be atomic